### ISSUE TITLE
Gun Variantion Suggestion

### ISSUE THREAD

**Issue by voidstonism** [Eggs-D-Studios/wos-issues#18]
### Guidelines

- [X] I checked for duplicate suggestions
- [X] My suggestion intends to add or improve something

### Version

Development (Unstable)

### Topic

A game mechanic (e.g. a part, a feature, etc)

### Why are you making your suggestion?

Energy Weapons are planned to have a wide diversity so why not kinetics too?

### What do you propose? What should change, or what should be added?

Guns should be more like a Template part where depending on what types of material is inserted it can change the properties of the gun They will also consume the same the part that was inserted into the gun e.i. (Gun inserted with titanium uses titanium except exotic matter) and of course they will be raytraced projectiles instead of raycast beams.

**Titanium:** Armor Piercing Gun, shoots a fast AP round that can damage multiple layers of iron and requires a a cool down of 4 secs between each shoot. To prevent spamming it also requires a 10-20 sec cooldown after 5 shots are fired. 
Uses 10-20 Titanium per shot

**Copper/Lead:** Machine Gun, a Rapidly firing gun that shoots weak projectiles used to kill players or weaker ships. has a cooldown of 0.2 sec between each shoot and a 10 sec cooldown after 50 shoots.
Uses 2 copper/lead per shot.

**Iron:** Shotgun, Close ranged weapon with a 3 sec reload in-between shots. 10 sec cooldown after 10 shots.
Uses 10 iron per shot.

**ExoticMatter:** Exoticmatter Gun, Powerful end game weapon which require a legnth of 20 studs long and 2 studs wide to fire and cannot be 50 studs near more than 3 other exoticmatter guns including it self.  Vaporizes layers of armor and can be partially reflected by the pllanned Mirror part.
3 Second Reload
Uses 5 DarkMatter to fire

**Comment by Minedwe-2** [comment]
I agree with this in concept, but I would propose expanding it some more (for example, weapon modules for kinetic weapons such as Autoloaders or other such Components once that system is released, and maybe the introduction of Chemical Energy rounds like HEAT that use more expensive ammo like Explosives but do very high damage to kinetic armor, but can be hard-countered with special armor types like Ceramic)

**Comment by Hexcede** [comment]
> so why not kinetics too?

I was actually looking to add new kinetic weapons as well, I just haven't had too many interesting ideas yet. Any weapons I add will not be part of 

A heat seeking flak weapon with a high firerrate and average DPS was one of the more interesting weapon ideas I had, with the intent behind it being that generally hot components inside of ships will be important ones that the flak would target. That would make the flak good at dealing targeted damage to important components in a ship. It would likely either be a semi-auto shotgun weapon or a full auto burst fire weapon.

The railgun rework was kinetic. The idea is that the rail the railgun shoots would be like wreathed in plasma and it'd be able to melt into ships (this would likely partially be how the pierce mechanics work for the railgun). It's sort of energy weapon ish but would be classified as a kinetic weapon and would mostly have the properties of a kinetic weapon.
